### PR TITLE
Add <VisibilityHelper /> component

### PR DESCRIPTION
**Pull request by adamziel**
## Description
While working on #20607, I noticed a use-case of applying `<VisuallyHidden />` conditionally. The cleanest way to do it is with a separate component that either creates `<VisuallyHidden />` if target should be hidden, or creates desired element otherwise. This adds `<VisibilityHelper />` component that does just that.

## How has this been tested?
Added snapshot tests.

For manual testing:

1. Open `packages/block-library/src/gallery/gallery.js`
1. Find `<RichText />` instance
1. Replace the declaration with `<VisibilityHelper as={ RichText } isHidden={ ! isSelected && RichText.isEmpty( caption ) } className="blocks-gallery-caption" /* ...other props */ />`
1. Observe that the `<VisuallyHidden />` is only applied when `isHidden` is true

## Types of changes
Bug fix (non-breaking change which fixes an issue)

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards.
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate.
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

File: packages/components/src/visually-hidden/index.js [hunk 1]
```diff
@@ -19,4 +19,26 @@ function VisuallyHidden( { as = 'div', className, ...props } ) {
 		...props,
 	} );
 }
+
+/**
+ * VisibilityHelper component to simplify common use-case of
+ * conditionally hiding components.
+ */
+export function VisibilityHelper( {
```
Comment (youknowriad) [inline packages/components/src/visually-hidden/index.js]: It's not clear to me why we need this component? It feels like something that should be done by the caller in general?

Comment (youknowriad) [reply packages/components/src/visually-hidden/index.js]: Also, why a new component, why not just a prop for VisuallyHidden

**Comment by gziolo** [comment]
I’m wondering we really need it in the first place. The example you shared might be a good candidate for a different  refactoring. The way the user interaction works made me think whether we could render nothing when the block isn’t selected and the caption is empty. I can’t think of the reason why it would need to be hidden visually if it can’t be focused when the block isn’t selected.

File: packages/components/src/visually-hidden/index.js (hunk 1 above)
Comment (adamziel) [reply packages/components/src/visually-hidden/index.js]: A prop was my first thought but I couldn't come up with a good name, e.g. all of these feels awkward: `<VisuallyHidden isHidden={true} />`, `<VisuallyHiddenMaybe isHidden={true} />`

**Comment by adamziel** [comment]
Thanks for feedback! I think we might end up doing this in caller in general, especially since there aren't that many instances.

>  I can’t think of the reason why it would need to be hidden visually if it can’t be focused when the block isn’t selected.

I assume it's to tell screen readers there is an editable label available in that block. That being said, we're not making it obvious for visual users, so maybe we could just remove it entirely as you suggested.

**Pull request closed without merging**